### DISCUSSION TITLE
Show timestamp and current running time in the logs?

### DISCUSSION THREAD

**Discussion by hverlin** [firecow/gitlab-ci-local#844] [Ideas]
Hi, I was wondering if you would be open adding an option to show the current time + runtime in the logs.
This is really useful for long-running jobs to get a better idea of how much time things have taken. Not all processes output the timestamps, unfortunately.

I could create a cleaner patch but for now, I just added a very simple change like this.
```ts
    get chalkJobName () { // rename
        const currentTime = new Date().toISOString().split("T")[1].split(".")[0];
        const runningFor = prettyHrtime(process.hrtime(this.startTime));

        // [16:33:19 (1.37 min)] my-job     > hello world
        return chalk`[${currentTime} (${runningFor})] {blueBright ${this.name.padEnd(this.jobNamePad)}}`;
    }
```

Let me know if you are interested in supporting it as an official option

(note that GitLab CI does not support this feature yet… https://gitlab.com/gitlab-org/gitlab/-/issues/202293)

**Comment by firecow** [comment]
I have thought about adding time to each line, but decided against it, because it make the lines quite long.

Having a `--timestamps` boolean cli option would be preferable, and would definently reach master 👍

**Reply by hverlin** [reply]
yeah, agree that it should not be there by default. Only useful for users who have long-running jobs
